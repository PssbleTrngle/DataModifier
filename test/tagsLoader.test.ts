import { createResolver } from '@pssbletrngle/pack-resolver'
import PackLoader from '../src/loader/pack'
import createTestLogger from './mock/TestLogger'

const logger = createTestLogger()
const loader = new PackLoader(logger)

beforeAll(async () => {
   const resolver = createResolver({ from: 'example', include: ['data/*/tags/**/*.json'] })
   await loader.loadFrom(resolver)
}, 10_000)

afterEach(() => {
   loader.clear()
})

test('loads tags correctly', async () => {
   const itemTags = loader.tagRegistry('items')
   const blockTags = loader.tagRegistry('blocks')

   expect(blockTags.list().length).toBe(259)
   expect(itemTags.list().length).toBe(302)

   expect(blockTags.get('minecraft:mineable/pickaxe')).toMatchSnapshot()
   expect(itemTags.get('minecraft:logs')).toMatchSnapshot()
})

test('resolves tags correctly', async () => {
   const itemTags = loader.tagRegistry('items')
   const blockTags = loader.tagRegistry('blocks')

   expect(blockTags.resolve('minecraft:mineable/axe')).toMatchSnapshot()
   expect(itemTags.resolve('minecraft:trapdoors')).toMatchSnapshot()
})
