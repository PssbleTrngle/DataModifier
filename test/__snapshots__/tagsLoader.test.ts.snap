// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loads tags correctly 1`] = `
[
  "create:schematicannon",
  "create:schematic_table",
  "create:shaft",
  "create:cogwheel",
  "create:large_cogwheel",
  "create:andesite_encased_shaft",
  "create:brass_encased_shaft",
  "create:andesite_encased_cogwheel",
  "create:brass_encased_cogwheel",
  "create:andesite_encased_large_cogwheel",
  "create:brass_encased_large_cogwheel",
  "create:gearbox",
  "create:clutch",
  "create:gearshift",
  "create:encased_chain_drive",
  "create:adjustable_chain_gearshift",
  "create:belt",
  "create:creative_motor",
  "create:water_wheel",
  "create:large_water_wheel",
  "create:water_wheel_structure",
  "create:encased_fan",
  "create:nozzle",
  "create:turntable",
  "create:hand_crank",
  "create:cuckoo_clock",
  "create:mysterious_cuckoo_clock",
  "create:millstone",
  "create:crushing_wheel",
  "create:mechanical_press",
  "create:mechanical_mixer",
  "create:basin",
  "create:blaze_burner",
  "create:lit_blaze_burner",
  "create:depot",
  "create:weighted_ejector",
  "create:chute",
  "create:smart_chute",
  "create:speedometer",
  "create:stressometer",
  "create:wooden_bracket",
  "create:metal_bracket",
  "create:fluid_pipe",
  "create:encased_fluid_pipe",
  "create:glass_fluid_pipe",
  "create:mechanical_pump",
  "create:smart_fluid_pipe",
  "create:fluid_valve",
  "create:copper_valve_handle",
  "create:white_valve_handle",
  "create:orange_valve_handle",
  "create:magenta_valve_handle",
  "create:light_blue_valve_handle",
  "create:yellow_valve_handle",
  "create:lime_valve_handle",
  "create:pink_valve_handle",
  "create:gray_valve_handle",
  "create:light_gray_valve_handle",
  "create:cyan_valve_handle",
  "create:purple_valve_handle",
  "create:blue_valve_handle",
  "create:brown_valve_handle",
  "create:green_valve_handle",
  "create:red_valve_handle",
  "create:black_valve_handle",
  "create:fluid_tank",
  "create:creative_fluid_tank",
  "create:hose_pulley",
  "create:item_drain",
  "create:spout",
  "create:portable_fluid_interface",
  "create:steam_engine",
  "create:steam_whistle",
  "create:steam_whistle_extension",
  "create:powered_shaft",
  "create:mechanical_piston",
  "create:sticky_mechanical_piston",
  "create:piston_extension_pole",
  "create:mechanical_piston_head",
  "create:gantry_carriage",
  "create:gantry_shaft",
  "create:windmill_bearing",
  "create:mechanical_bearing",
  "create:clockwork_bearing",
  "create:rope_pulley",
  "create:elevator_pulley",
  "create:cart_assembler",
  "create:controller_rail",
  "create:linear_chassis",
  "create:secondary_linear_chassis",
  "create:radial_chassis",
  "create:sticker",
  "create:contraption_controls",
  "create:mechanical_drill",
  "create:mechanical_saw",
  "create:deployer",
  "create:portable_storage_interface",
  "create:redstone_contact",
  "create:elevator_contact",
  "create:mechanical_harvester",
  "create:mechanical_plough",
  "create:mechanical_roller",
  "create:andesite_casing",
  "create:brass_casing",
  "create:copper_casing",
  "create:shadow_steel_casing",
  "create:refined_radiance_casing",
  "create:mechanical_crafter",
  "create:sequenced_gearshift",
  "create:flywheel",
  "create:rotation_speed_controller",
  "create:mechanical_arm",
  "create:track",
  "create:railway_casing",
  "create:track_station",
  "create:track_signal",
  "create:track_observer",
  "create:small_bogey",
  "create:large_bogey",
  "create:controls",
  "create:item_vault",
  "create:andesite_funnel",
  "create:andesite_belt_funnel",
  "create:brass_funnel",
  "create:brass_belt_funnel",
  "create:andesite_tunnel",
  "create:brass_tunnel",
  "create:content_observer",
  "create:stockpile_switch",
  "create:creative_crate",
  "create:display_link",
  "create:display_board",
  "create:nixie_tube",
  "create:white_nixie_tube",
  "create:magenta_nixie_tube",
  "create:light_blue_nixie_tube",
  "create:yellow_nixie_tube",
  "create:lime_nixie_tube",
  "create:pink_nixie_tube",
  "create:gray_nixie_tube",
  "create:light_gray_nixie_tube",
  "create:cyan_nixie_tube",
  "create:purple_nixie_tube",
  "create:blue_nixie_tube",
  "create:brown_nixie_tube",
  "create:green_nixie_tube",
  "create:red_nixie_tube",
  "create:black_nixie_tube",
  "create:rose_quartz_lamp",
  "create:redstone_link",
  "create:analog_lever",
  "create:placard",
  "create:copper_backtank",
  "create:netherite_backtank",
  "create:peculiar_bell",
  "create:haunted_bell",
  "create:clipboard",
  "create:andesite_ladder",
  "create:brass_ladder",
  "create:copper_ladder",
  "create:andesite_bars",
  "create:brass_bars",
  "create:copper_bars",
  "create:andesite_scaffolding",
  "create:brass_scaffolding",
  "create:copper_scaffolding",
  "create:metal_girder",
  "create:metal_girder_encased_shaft",
  "create:copycat_base",
  "create:copycat_step",
  "create:copycat_panel",
  "create:andesite_door",
  "create:brass_door",
  "create:copper_door",
  "create:train_door",
  "create:train_trapdoor",
  "create:framed_glass_door",
  "create:framed_glass_trapdoor",
  "create:zinc_ore",
  "create:deepslate_zinc_ore",
  "create:raw_zinc_block",
  "create:zinc_block",
  "create:andesite_alloy_block",
  "create:industrial_iron_block",
  "create:brass_block",
  "create:experience_block",
  "create:rose_quartz_block",
  "create:rose_quartz_tiles",
  "create:small_rose_quartz_tiles",
  "create:copper_shingles",
  "create:exposed_copper_shingles",
  "create:weathered_copper_shingles",
  "create:oxidized_copper_shingles",
  "create:copper_shingle_slab",
  "create:exposed_copper_shingle_slab",
  "create:weathered_copper_shingle_slab",
  "create:oxidized_copper_shingle_slab",
  "create:copper_shingle_stairs",
  "create:exposed_copper_shingle_stairs",
  "create:weathered_copper_shingle_stairs",
  "create:oxidized_copper_shingle_stairs",
  "create:waxed_copper_shingles",
  "create:waxed_exposed_copper_shingles",
  "create:waxed_weathered_copper_shingles",
  "create:waxed_oxidized_copper_shingles",
  "create:waxed_copper_shingle_slab",
  "create:waxed_exposed_copper_shingle_slab",
  "create:waxed_weathered_copper_shingle_slab",
  "create:waxed_oxidized_copper_shingle_slab",
  "create:waxed_copper_shingle_stairs",
  "create:waxed_exposed_copper_shingle_stairs",
  "create:waxed_weathered_copper_shingle_stairs",
  "create:waxed_oxidized_copper_shingle_stairs",
  "create:copper_tiles",
  "create:exposed_copper_tiles",
  "create:weathered_copper_tiles",
  "create:oxidized_copper_tiles",
  "create:copper_tile_slab",
  "create:exposed_copper_tile_slab",
  "create:weathered_copper_tile_slab",
  "create:oxidized_copper_tile_slab",
  "create:copper_tile_stairs",
  "create:exposed_copper_tile_stairs",
  "create:weathered_copper_tile_stairs",
  "create:oxidized_copper_tile_stairs",
  "create:waxed_copper_tiles",
  "create:waxed_exposed_copper_tiles",
  "create:waxed_weathered_copper_tiles",
  "create:waxed_oxidized_copper_tiles",
  "create:waxed_copper_tile_slab",
  "create:waxed_exposed_copper_tile_slab",
  "create:waxed_weathered_copper_tile_slab",
  "create:waxed_oxidized_copper_tile_slab",
  "create:waxed_copper_tile_stairs",
  "create:waxed_exposed_copper_tile_stairs",
  "create:waxed_weathered_copper_tile_stairs",
  "create:waxed_oxidized_copper_tile_stairs",
  "create:cut_granite",
  "create:cut_granite_stairs",
  "create:cut_granite_slab",
  "create:cut_granite_wall",
  "create:polished_cut_granite",
  "create:polished_cut_granite_stairs",
  "create:polished_cut_granite_slab",
  "create:polished_cut_granite_wall",
  "create:cut_granite_bricks",
  "create:cut_granite_brick_stairs",
  "create:cut_granite_brick_slab",
  "create:cut_granite_brick_wall",
  "create:small_granite_bricks",
  "create:small_granite_brick_stairs",
  "create:small_granite_brick_slab",
  "create:small_granite_brick_wall",
  "create:layered_granite",
  "create:granite_pillar",
  "create:cut_diorite",
  "create:cut_diorite_stairs",
  "create:cut_diorite_slab",
  "create:cut_diorite_wall",
  "create:polished_cut_diorite",
  "create:polished_cut_diorite_stairs",
  "create:polished_cut_diorite_slab",
  "create:polished_cut_diorite_wall",
  "create:cut_diorite_bricks",
  "create:cut_diorite_brick_stairs",
  "create:cut_diorite_brick_slab",
  "create:cut_diorite_brick_wall",
  "create:small_diorite_bricks",
  "create:small_diorite_brick_stairs",
  "create:small_diorite_brick_slab",
  "create:small_diorite_brick_wall",
  "create:layered_diorite",
  "create:diorite_pillar",
  "create:cut_andesite",
  "create:cut_andesite_stairs",
  "create:cut_andesite_slab",
  "create:cut_andesite_wall",
  "create:polished_cut_andesite",
  "create:polished_cut_andesite_stairs",
  "create:polished_cut_andesite_slab",
  "create:polished_cut_andesite_wall",
  "create:cut_andesite_bricks",
  "create:cut_andesite_brick_stairs",
  "create:cut_andesite_brick_slab",
  "create:cut_andesite_brick_wall",
  "create:small_andesite_bricks",
  "create:small_andesite_brick_stairs",
  "create:small_andesite_brick_slab",
  "create:small_andesite_brick_wall",
  "create:layered_andesite",
  "create:andesite_pillar",
  "create:cut_calcite",
  "create:cut_calcite_stairs",
  "create:cut_calcite_slab",
  "create:cut_calcite_wall",
  "create:polished_cut_calcite",
  "create:polished_cut_calcite_stairs",
  "create:polished_cut_calcite_slab",
  "create:polished_cut_calcite_wall",
  "create:cut_calcite_bricks",
  "create:cut_calcite_brick_stairs",
  "create:cut_calcite_brick_slab",
  "create:cut_calcite_brick_wall",
  "create:small_calcite_bricks",
  "create:small_calcite_brick_stairs",
  "create:small_calcite_brick_slab",
  "create:small_calcite_brick_wall",
  "create:layered_calcite",
  "create:calcite_pillar",
  "create:cut_dripstone",
  "create:cut_dripstone_stairs",
  "create:cut_dripstone_slab",
  "create:cut_dripstone_wall",
  "create:polished_cut_dripstone",
  "create:polished_cut_dripstone_stairs",
  "create:polished_cut_dripstone_slab",
  "create:polished_cut_dripstone_wall",
  "create:cut_dripstone_bricks",
  "create:cut_dripstone_brick_stairs",
  "create:cut_dripstone_brick_slab",
  "create:cut_dripstone_brick_wall",
  "create:small_dripstone_bricks",
  "create:small_dripstone_brick_stairs",
  "create:small_dripstone_brick_slab",
  "create:small_dripstone_brick_wall",
  "create:layered_dripstone",
  "create:dripstone_pillar",
  "create:cut_deepslate",
  "create:cut_deepslate_stairs",
  "create:cut_deepslate_slab",
  "create:cut_deepslate_wall",
  "create:polished_cut_deepslate",
  "create:polished_cut_deepslate_stairs",
  "create:polished_cut_deepslate_slab",
  "create:polished_cut_deepslate_wall",
  "create:cut_deepslate_bricks",
  "create:cut_deepslate_brick_stairs",
  "create:cut_deepslate_brick_slab",
  "create:cut_deepslate_brick_wall",
  "create:small_deepslate_bricks",
  "create:small_deepslate_brick_stairs",
  "create:small_deepslate_brick_slab",
  "create:small_deepslate_brick_wall",
  "create:layered_deepslate",
  "create:deepslate_pillar",
  "create:cut_tuff",
  "create:cut_tuff_stairs",
  "create:cut_tuff_slab",
  "create:cut_tuff_wall",
  "create:polished_cut_tuff",
  "create:polished_cut_tuff_stairs",
  "create:polished_cut_tuff_slab",
  "create:polished_cut_tuff_wall",
  "create:cut_tuff_bricks",
  "create:cut_tuff_brick_stairs",
  "create:cut_tuff_brick_slab",
  "create:cut_tuff_brick_wall",
  "create:small_tuff_bricks",
  "create:small_tuff_brick_stairs",
  "create:small_tuff_brick_slab",
  "create:small_tuff_brick_wall",
  "create:layered_tuff",
  "create:tuff_pillar",
  "create:asurine",
  "create:cut_asurine",
  "create:cut_asurine_stairs",
  "create:cut_asurine_slab",
  "create:cut_asurine_wall",
  "create:polished_cut_asurine",
  "create:polished_cut_asurine_stairs",
  "create:polished_cut_asurine_slab",
  "create:polished_cut_asurine_wall",
  "create:cut_asurine_bricks",
  "create:cut_asurine_brick_stairs",
  "create:cut_asurine_brick_slab",
  "create:cut_asurine_brick_wall",
  "create:small_asurine_bricks",
  "create:small_asurine_brick_stairs",
  "create:small_asurine_brick_slab",
  "create:small_asurine_brick_wall",
  "create:layered_asurine",
  "create:asurine_pillar",
  "create:crimsite",
  "create:cut_crimsite",
  "create:cut_crimsite_stairs",
  "create:cut_crimsite_slab",
  "create:cut_crimsite_wall",
  "create:polished_cut_crimsite",
  "create:polished_cut_crimsite_stairs",
  "create:polished_cut_crimsite_slab",
  "create:polished_cut_crimsite_wall",
  "create:cut_crimsite_bricks",
  "create:cut_crimsite_brick_stairs",
  "create:cut_crimsite_brick_slab",
  "create:cut_crimsite_brick_wall",
  "create:small_crimsite_bricks",
  "create:small_crimsite_brick_stairs",
  "create:small_crimsite_brick_slab",
  "create:small_crimsite_brick_wall",
  "create:layered_crimsite",
  "create:crimsite_pillar",
  "create:limestone",
  "create:cut_limestone",
  "create:cut_limestone_stairs",
  "create:cut_limestone_slab",
  "create:cut_limestone_wall",
  "create:polished_cut_limestone",
  "create:polished_cut_limestone_stairs",
  "create:polished_cut_limestone_slab",
  "create:polished_cut_limestone_wall",
  "create:cut_limestone_bricks",
  "create:cut_limestone_brick_stairs",
  "create:cut_limestone_brick_slab",
  "create:cut_limestone_brick_wall",
  "create:small_limestone_bricks",
  "create:small_limestone_brick_stairs",
  "create:small_limestone_brick_slab",
  "create:small_limestone_brick_wall",
  "create:layered_limestone",
  "create:limestone_pillar",
  "create:ochrum",
  "create:cut_ochrum",
  "create:cut_ochrum_stairs",
  "create:cut_ochrum_slab",
  "create:cut_ochrum_wall",
  "create:polished_cut_ochrum",
  "create:polished_cut_ochrum_stairs",
  "create:polished_cut_ochrum_slab",
  "create:polished_cut_ochrum_wall",
  "create:cut_ochrum_bricks",
  "create:cut_ochrum_brick_stairs",
  "create:cut_ochrum_brick_slab",
  "create:cut_ochrum_brick_wall",
  "create:small_ochrum_bricks",
  "create:small_ochrum_brick_stairs",
  "create:small_ochrum_brick_slab",
  "create:small_ochrum_brick_wall",
  "create:layered_ochrum",
  "create:ochrum_pillar",
  "create:scoria",
  "create:cut_scoria",
  "create:cut_scoria_stairs",
  "create:cut_scoria_slab",
  "create:cut_scoria_wall",
  "create:polished_cut_scoria",
  "create:polished_cut_scoria_stairs",
  "create:polished_cut_scoria_slab",
  "create:polished_cut_scoria_wall",
  "create:cut_scoria_bricks",
  "create:cut_scoria_brick_stairs",
  "create:cut_scoria_brick_slab",
  "create:cut_scoria_brick_wall",
  "create:small_scoria_bricks",
  "create:small_scoria_brick_stairs",
  "create:small_scoria_brick_slab",
  "create:small_scoria_brick_wall",
  "create:layered_scoria",
  "create:scoria_pillar",
  "create:scorchia",
  "create:cut_scorchia",
  "create:cut_scorchia_stairs",
  "create:cut_scorchia_slab",
  "create:cut_scorchia_wall",
  "create:polished_cut_scorchia",
  "create:polished_cut_scorchia_stairs",
  "create:polished_cut_scorchia_slab",
  "create:polished_cut_scorchia_wall",
  "create:cut_scorchia_bricks",
  "create:cut_scorchia_brick_stairs",
  "create:cut_scorchia_brick_slab",
  "create:cut_scorchia_brick_wall",
  "create:small_scorchia_bricks",
  "create:small_scorchia_brick_stairs",
  "create:small_scorchia_brick_slab",
  "create:small_scorchia_brick_wall",
  "create:layered_scorchia",
  "create:scorchia_pillar",
  "create:veridium",
  "create:cut_veridium",
  "create:cut_veridium_stairs",
  "create:cut_veridium_slab",
  "create:cut_veridium_wall",
  "create:polished_cut_veridium",
  "create:polished_cut_veridium_stairs",
  "create:polished_cut_veridium_slab",
  "create:polished_cut_veridium_wall",
  "create:cut_veridium_bricks",
  "create:cut_veridium_brick_stairs",
  "create:cut_veridium_brick_slab",
  "create:cut_veridium_brick_wall",
  "create:small_veridium_bricks",
  "create:small_veridium_brick_stairs",
  "create:small_veridium_brick_slab",
  "create:small_veridium_brick_wall",
  "create:layered_veridium",
  "create:veridium_pillar",
  "minecraft:stone",
  "minecraft:granite",
  "minecraft:polished_granite",
  "minecraft:diorite",
  "minecraft:polished_diorite",
  "minecraft:andesite",
  "minecraft:polished_andesite",
  "minecraft:cobblestone",
  "minecraft:gold_ore",
  "minecraft:deepslate_gold_ore",
  "minecraft:iron_ore",
  "minecraft:deepslate_iron_ore",
  "minecraft:coal_ore",
  "minecraft:deepslate_coal_ore",
  "minecraft:nether_gold_ore",
  "minecraft:lapis_ore",
  "minecraft:deepslate_lapis_ore",
  "minecraft:lapis_block",
  "minecraft:dispenser",
  "minecraft:sandstone",
  "minecraft:chiseled_sandstone",
  "minecraft:cut_sandstone",
  "minecraft:gold_block",
  "minecraft:iron_block",
  "minecraft:bricks",
  "minecraft:mossy_cobblestone",
  "minecraft:obsidian",
  "minecraft:spawner",
  "minecraft:diamond_ore",
  "minecraft:deepslate_diamond_ore",
  "minecraft:diamond_block",
  "minecraft:furnace",
  "minecraft:cobblestone_stairs",
  "minecraft:stone_pressure_plate",
  "minecraft:iron_door",
  "minecraft:redstone_ore",
  "minecraft:deepslate_redstone_ore",
  "minecraft:netherrack",
  "minecraft:basalt",
  "minecraft:polished_basalt",
  "minecraft:stone_bricks",
  "minecraft:mossy_stone_bricks",
  "minecraft:cracked_stone_bricks",
  "minecraft:chiseled_stone_bricks",
  "minecraft:iron_bars",
  "minecraft:chain",
  "minecraft:brick_stairs",
  "minecraft:stone_brick_stairs",
  "minecraft:nether_bricks",
  "minecraft:nether_brick_fence",
  "minecraft:nether_brick_stairs",
  "minecraft:enchanting_table",
  "minecraft:brewing_stand",
  "minecraft:end_stone",
  "minecraft:sandstone_stairs",
  "minecraft:emerald_ore",
  "minecraft:deepslate_emerald_ore",
  "minecraft:ender_chest",
  "minecraft:emerald_block",
  "minecraft:light_weighted_pressure_plate",
  "minecraft:heavy_weighted_pressure_plate",
  "minecraft:redstone_block",
  "minecraft:nether_quartz_ore",
  "minecraft:hopper",
  "minecraft:quartz_block",
  "minecraft:chiseled_quartz_block",
  "minecraft:quartz_pillar",
  "minecraft:quartz_stairs",
  "minecraft:dropper",
  "minecraft:white_terracotta",
  "minecraft:orange_terracotta",
  "minecraft:magenta_terracotta",
  "minecraft:light_blue_terracotta",
  "minecraft:yellow_terracotta",
  "minecraft:lime_terracotta",
  "minecraft:pink_terracotta",
  "minecraft:gray_terracotta",
  "minecraft:light_gray_terracotta",
  "minecraft:cyan_terracotta",
  "minecraft:purple_terracotta",
  "minecraft:blue_terracotta",
  "minecraft:brown_terracotta",
  "minecraft:green_terracotta",
  "minecraft:red_terracotta",
  "minecraft:black_terracotta",
  "minecraft:iron_trapdoor",
  "minecraft:prismarine",
  "minecraft:prismarine_bricks",
  "minecraft:dark_prismarine",
  "minecraft:prismarine_stairs",
  "minecraft:prismarine_brick_stairs",
  "minecraft:dark_prismarine_stairs",
  "minecraft:prismarine_slab",
  "minecraft:prismarine_brick_slab",
  "minecraft:dark_prismarine_slab",
  "minecraft:terracotta",
  "minecraft:coal_block",
  "minecraft:red_sandstone",
  "minecraft:chiseled_red_sandstone",
  "minecraft:cut_red_sandstone",
  "minecraft:red_sandstone_stairs",
  "minecraft:stone_slab",
  "minecraft:smooth_stone_slab",
  "minecraft:sandstone_slab",
  "minecraft:cut_sandstone_slab",
  "minecraft:petrified_oak_slab",
  "minecraft:cobblestone_slab",
  "minecraft:brick_slab",
  "minecraft:stone_brick_slab",
  "minecraft:nether_brick_slab",
  "minecraft:quartz_slab",
  "minecraft:red_sandstone_slab",
  "minecraft:cut_red_sandstone_slab",
  "minecraft:purpur_slab",
  "minecraft:smooth_stone",
  "minecraft:smooth_sandstone",
  "minecraft:smooth_quartz",
  "minecraft:smooth_red_sandstone",
  "minecraft:purpur_block",
  "minecraft:purpur_pillar",
  "minecraft:purpur_stairs",
  "minecraft:end_stone_bricks",
  "minecraft:magma_block",
  "minecraft:red_nether_bricks",
  "minecraft:bone_block",
  "minecraft:observer",
  "minecraft:white_glazed_terracotta",
  "minecraft:orange_glazed_terracotta",
  "minecraft:magenta_glazed_terracotta",
  "minecraft:light_blue_glazed_terracotta",
  "minecraft:yellow_glazed_terracotta",
  "minecraft:lime_glazed_terracotta",
  "minecraft:pink_glazed_terracotta",
  "minecraft:gray_glazed_terracotta",
  "minecraft:light_gray_glazed_terracotta",
  "minecraft:cyan_glazed_terracotta",
  "minecraft:purple_glazed_terracotta",
  "minecraft:blue_glazed_terracotta",
  "minecraft:brown_glazed_terracotta",
  "minecraft:green_glazed_terracotta",
  "minecraft:red_glazed_terracotta",
  "minecraft:black_glazed_terracotta",
  "minecraft:white_concrete",
  "minecraft:orange_concrete",
  "minecraft:magenta_concrete",
  "minecraft:light_blue_concrete",
  "minecraft:yellow_concrete",
  "minecraft:lime_concrete",
  "minecraft:pink_concrete",
  "minecraft:gray_concrete",
  "minecraft:light_gray_concrete",
  "minecraft:cyan_concrete",
  "minecraft:purple_concrete",
  "minecraft:blue_concrete",
  "minecraft:brown_concrete",
  "minecraft:green_concrete",
  "minecraft:red_concrete",
  "minecraft:black_concrete",
  "minecraft:dead_tube_coral_block",
  "minecraft:dead_brain_coral_block",
  "minecraft:dead_bubble_coral_block",
  "minecraft:dead_fire_coral_block",
  "minecraft:dead_horn_coral_block",
  "minecraft:tube_coral_block",
  "minecraft:brain_coral_block",
  "minecraft:bubble_coral_block",
  "minecraft:fire_coral_block",
  "minecraft:horn_coral_block",
  "minecraft:dead_tube_coral",
  "minecraft:dead_brain_coral",
  "minecraft:dead_bubble_coral",
  "minecraft:dead_fire_coral",
  "minecraft:dead_horn_coral",
  "minecraft:dead_tube_coral_fan",
  "minecraft:dead_brain_coral_fan",
  "minecraft:dead_bubble_coral_fan",
  "minecraft:dead_fire_coral_fan",
  "minecraft:dead_horn_coral_fan",
  "minecraft:dead_tube_coral_wall_fan",
  "minecraft:dead_brain_coral_wall_fan",
  "minecraft:dead_bubble_coral_wall_fan",
  "minecraft:dead_fire_coral_wall_fan",
  "minecraft:dead_horn_coral_wall_fan",
  "minecraft:polished_granite_stairs",
  "minecraft:smooth_red_sandstone_stairs",
  "minecraft:mossy_stone_brick_stairs",
  "minecraft:polished_diorite_stairs",
  "minecraft:mossy_cobblestone_stairs",
  "minecraft:end_stone_brick_stairs",
  "minecraft:stone_stairs",
  "minecraft:smooth_sandstone_stairs",
  "minecraft:smooth_quartz_stairs",
  "minecraft:granite_stairs",
  "minecraft:andesite_stairs",
  "minecraft:red_nether_brick_stairs",
  "minecraft:polished_andesite_stairs",
  "minecraft:diorite_stairs",
  "minecraft:polished_granite_slab",
  "minecraft:smooth_red_sandstone_slab",
  "minecraft:mossy_stone_brick_slab",
  "minecraft:polished_diorite_slab",
  "minecraft:mossy_cobblestone_slab",
  "minecraft:end_stone_brick_slab",
  "minecraft:smooth_sandstone_slab",
  "minecraft:smooth_quartz_slab",
  "minecraft:granite_slab",
  "minecraft:andesite_slab",
  "minecraft:red_nether_brick_slab",
  "minecraft:polished_andesite_slab",
  "minecraft:diorite_slab",
  "minecraft:smoker",
  "minecraft:blast_furnace",
  "minecraft:grindstone",
  "minecraft:stonecutter",
  "minecraft:bell",
  "minecraft:lantern",
  "minecraft:soul_lantern",
  "minecraft:warped_nylium",
  "minecraft:crimson_nylium",
  "minecraft:netherite_block",
  "minecraft:ancient_debris",
  "minecraft:crying_obsidian",
  "minecraft:respawn_anchor",
  "minecraft:lodestone",
  "minecraft:blackstone",
  "minecraft:blackstone_stairs",
  "minecraft:blackstone_slab",
  "minecraft:polished_blackstone",
  "minecraft:polished_blackstone_bricks",
  "minecraft:cracked_polished_blackstone_bricks",
  "minecraft:chiseled_polished_blackstone",
  "minecraft:polished_blackstone_brick_slab",
  "minecraft:polished_blackstone_brick_stairs",
  "minecraft:gilded_blackstone",
  "minecraft:polished_blackstone_stairs",
  "minecraft:polished_blackstone_slab",
  "minecraft:polished_blackstone_pressure_plate",
  "minecraft:chiseled_nether_bricks",
  "minecraft:cracked_nether_bricks",
  "minecraft:quartz_bricks",
  "minecraft:tuff",
  "minecraft:calcite",
  "minecraft:oxidized_copper",
  "minecraft:weathered_copper",
  "minecraft:exposed_copper",
  "minecraft:copper_block",
  "minecraft:copper_ore",
  "minecraft:deepslate_copper_ore",
  "minecraft:oxidized_cut_copper",
  "minecraft:weathered_cut_copper",
  "minecraft:exposed_cut_copper",
  "minecraft:cut_copper",
  "minecraft:oxidized_cut_copper_stairs",
  "minecraft:weathered_cut_copper_stairs",
  "minecraft:exposed_cut_copper_stairs",
  "minecraft:cut_copper_stairs",
  "minecraft:oxidized_cut_copper_slab",
  "minecraft:weathered_cut_copper_slab",
  "minecraft:exposed_cut_copper_slab",
  "minecraft:cut_copper_slab",
  "minecraft:waxed_copper_block",
  "minecraft:waxed_weathered_copper",
  "minecraft:waxed_exposed_copper",
  "minecraft:waxed_oxidized_copper",
  "minecraft:waxed_oxidized_cut_copper",
  "minecraft:waxed_weathered_cut_copper",
  "minecraft:waxed_exposed_cut_copper",
  "minecraft:waxed_cut_copper",
  "minecraft:waxed_oxidized_cut_copper_stairs",
  "minecraft:waxed_weathered_cut_copper_stairs",
  "minecraft:waxed_exposed_cut_copper_stairs",
  "minecraft:waxed_cut_copper_stairs",
  "minecraft:waxed_oxidized_cut_copper_slab",
  "minecraft:waxed_weathered_cut_copper_slab",
  "minecraft:waxed_exposed_cut_copper_slab",
  "minecraft:waxed_cut_copper_slab",
  "minecraft:lightning_rod",
  "minecraft:pointed_dripstone",
  "minecraft:dripstone_block",
  "minecraft:deepslate",
  "minecraft:cobbled_deepslate",
  "minecraft:cobbled_deepslate_stairs",
  "minecraft:cobbled_deepslate_slab",
  "minecraft:polished_deepslate",
  "minecraft:polished_deepslate_stairs",
  "minecraft:polished_deepslate_slab",
  "minecraft:deepslate_tiles",
  "minecraft:deepslate_tile_stairs",
  "minecraft:deepslate_tile_slab",
  "minecraft:deepslate_bricks",
  "minecraft:deepslate_brick_stairs",
  "minecraft:deepslate_brick_slab",
  "minecraft:chiseled_deepslate",
  "minecraft:cracked_deepslate_bricks",
  "minecraft:cracked_deepslate_tiles",
  "minecraft:smooth_basalt",
  "minecraft:raw_iron_block",
  "minecraft:raw_copper_block",
  "minecraft:raw_gold_block",
  "minecraft:ice",
  "minecraft:packed_ice",
  "minecraft:blue_ice",
  "minecraft:stone_button",
  "minecraft:piston",
  "minecraft:sticky_piston",
  "minecraft:piston_head",
  "minecraft:amethyst_cluster",
  "minecraft:small_amethyst_bud",
  "minecraft:medium_amethyst_bud",
  "minecraft:large_amethyst_bud",
  "minecraft:amethyst_block",
  "minecraft:budding_amethyst",
  "minecraft:infested_cobblestone",
  "minecraft:infested_chiseled_stone_bricks",
  "minecraft:infested_cracked_stone_bricks",
  "minecraft:infested_deepslate",
  "minecraft:infested_stone",
  "minecraft:infested_mossy_stone_bricks",
  "minecraft:infested_stone_bricks",
  "#minecraft:walls",
  "#minecraft:shulker_boxes",
  "#minecraft:anvil",
  "#minecraft:cauldrons",
  "#minecraft:rails",
  "minecraft:conduit",
]
`;

exports[`loads tags correctly 2`] = `
[
  "#minecraft:logs_that_burn",
  "#minecraft:crimson_stems",
  "#minecraft:warped_stems",
]
`;
